tardis/util/base.py

```diff
@@ -596,3 +596,15 @@ def is_notebook():
     # All other shell instances are returned False
     else:
         return False
+
+class QDataFrame(pd.DataFrame):
+    # normal properties
+    _metadata = ["units"]
+    def __init__(self, *args, **kwargs):
+        units = kwargs.pop('units', [])
```

Comment:
Should we require the length of the units array to be equal to the number of columns? Would it be more useful to have the units attribute be a dictionary with (key,value) pairs equal to (df_column_name, unit)?